Add back content to persistent index (#925)
arangodb/docs: 3.10/indexing-index-basics.md

indexes, indexing the `_from` and `_to` attributes of edge collections.
They provide very quick access to all edges originating in or arriving
at a given vertex, which allows to quickly find all neighbors of a vertex
in a graph.

In many cases one would like to run more specific queries, for example
finding amongst the edges originating from a given vertex only those
with a timestamp greater than or equal to some date and time. Exactly this
is achieved with "vertex-centric indexes". In a sense these are localized
indexes for an edge collection, which sit at every single vertex.

Technically, they are implemented in ArangoDB as indexes, which sort the 
complete edge collection first by `_from` and then by other attributes
for _OUTBOUND_ traversals, or first by `_to` and then by other attributes
for _INBOUND_ traversals. For traversals in _ANY_ direction two indexes
are needed, one with `_from` and the other with `_to` as first indexed field.

If we for example have a persistent index on the attributes `_from` and 
`timestamp` of an edge collection, we can answer the above question
very quickly with a single range lookup in the index.

> You can create sorted persistent indexes that index the special edge attributes

Reviewer: ```suggestion
You can create sorted persistent indexes that indexes the special edge attributes
```

Author: "index the special edge attributes" (plural) seems correct?

`_from` or `_to` and additionally other attributes. These are used
in graph traversals, when appropriate `FILTER` statements are found
by the optimizer.

For example, to create a vertex-centric index of the above type, you 
would simply do

```js
db.edges.ensureIndex({"type":"persistent", "fields": ["_from", "timestamp"]});
```

in arangosh. Then, queries like

```js
FOR v, e, p IN 1..1 OUTBOUND "V/1" edges
  FILTER e.timestamp >= "2018-07-09"
  RETURN p
```

will be considerably faster in case there are many edges originating

> from vertex `"V/1"` but only few with a recent time stamp. Note that the

Reviewer: ```suggestion
from vertex `"V/1"` but only few with a recent timestamp. Note that the
```

  created, by adding a combined index for the fields `_from` and `_to` to an edge collection.

- **unique, sparse persistent index**: all documents in the collection must have different
  values for the attributes covered by the unique index. Documents in which at least one
  of the index attributes is not set or has a value of `null` are not included in the
  index. This type of index can be used to ensure that there are no duplicate keys in
  the collection for documents that have the indexed attributes set. As the index will
  exclude documents for which the indexed attributes are `null` or not set, it can be
  used for optional attributes.

- **non-unique persistent index**: all documents in the collection will be indexed. This type
  of index is not sparse. Documents that do not contain the index attributes or that have
  a value of `null` in the index attribute(s) will still be indexed. Duplicate key values
  can occur and do not lead to unique constraint violations.

- **non-unique, sparse persistent index**: only those documents will be indexed that have all
  the indexed attributes set to a value other than `null`. It can be used for optional
  attributes.

> If a persistent index is created on an attribute that is missing in all or many

Reviewer: Covered in the sections before already. I suggest removing this to shorten the section!

comparison operators (`==`, `!=`, `>`, `>=`, `<`, `<=`, `ANY`, `ALL`, `NONE`)
cannot use array indexes currently.

> Ensure uniqueness of relations in edge collections
> --------------------------------------------------

Reviewer: "Ensure ..." sounds a bit like a command.
```suggestion
Ensuring uniqueness of relations in edge collections
----------------------------------------------------
```

You can create a combined index over the edge attributes `_from` and `_to`

> with the unique option enabled to prevent duplicate relations from being created.

Author: It's a specific use case (graph) that comes up often, so we added a dedicated section that we can point users to.

For example, a document collection *verts* might contain vertices with the document
handles `verts/A`, `verts/B` and `verts/C`. Relations between these documents can
be stored in an edge collection *edges* for instance. Now, you may want to make sure
that the vertex `verts/A` is never linked to `verts/B` by an edge more than once.
This can be achieved by adding a unique, non-sparse persistent index for the

> fields `_from` and `_to`:

Author: It's a specific use case (graph) that comes up often, so we added a dedicated section that we can point users to.
